removing jQuery dependency

Xirdion opened the pull request:
With this PR the bundle could now be used without jQuery.
To be able to get the parameters for each slider, I created a JS-variable `sliderConfig`.
The swiper-script and the contao-swiper-script are loaded at the end of the body-tag (`TL_BODY`) so that the `sliderConfig`-variable is already set, when the script needs it.

fritzmg commented:
Yes, I already did 15 days ago ;)

Xirdion commented:
Okay cool.
So should i change some things or are you planning to merge this in your bundle?

fritzmg commented:
You need to make some changes according to the comments I left in the review.

Xirdion commented:
Sorry, but I think I have a problem. I can not see any review you did.
Do I have to change some settings or something like that?

fritzmg commented:
@Xirdion ah sorry, I didn't actually submit the review request! You should be able to see it now.

Xirdion commented:
@fritzmg Now I can see them :)
I will go through your change requests as soon as possible.

Xirdion commented on src/Resources/public/contao-swiper.js:
```diff
@@ -0,0 +1,51 @@
+            if (cssClass) {
+                // add custom css class
+                wrapperElement.classList.add(cssClass);
+            }
```
Now I get the intention of this.
Now, when there is a value for the custom-wrapper class, I check if the next element inside the slider-container has this class.
This element will then be the slider-wrapper.

fritzmg replied:
It does not necessarily have to be the next element. It can be nested deeper too. My original jQuery code was this:
```php
<?php if (!$this->wrapperClass): ?>
  $('<?= $this->container ?> > *').wrapAll($('<div class="swiper-wrapper"></div>'));
<?php else: ?>
  $('<?= $this->container ?> .<?= $this->wrapperClass ?>').addClass('swiper-wrapper');
<?php endif; ?>
```
The wrapper class can be _anywhere within the container_. This needs to be kept the same, otherwise it breaks the backwards compatibility.

Xirdion replied:
So now the element with the given class is searched inside the container. The first occurrence will be used. It does not need to be the very next element.

fritzmg replied:
👍 looks good :)

Xirdion commented on src/Resources/public/contao-swiper.js:
```diff
@@ -0,0 +1,51 @@
+            if (params.navigation) {
+                params.navigation = {
+                    nextEl: '.swiper-button-next',
+                    prevEl: '.swiper-button-prev',
```
Yes this works. I just use it the way it is described in the documentation
http://idangero.us/swiper/api/
![screenshot from 2019-02-15 11-13-36](https://user-images.githubusercontent.com/17783055/52850034-d0935680-3112-11e9-8df0-0ace2647c5e0.png)

fritzmg replied:
Does it also work with
```
"use strict";
```
?

Xirdion replied:
Yes it also works with `"use strict"`. I added the line at the beginning of the code. Is that correct?

fritzmg replied:
No, it should be within the (function()) wrapper - but it's fine, I was just wondering. You do not have to add it.

Xirdion commented on src/Resources/public/contao-swiper.js:
```diff
@@ -0,0 +1,51 @@
+
+            // adding lazy.loading
+            params.lazy = {
+                loadPrevNext: true,
```
Here is the same as above.
In the documentation it is described this way.

Xirdion commented on src/Resources/public/element.css:
```diff
@@ -1,3 +1,4 @@
 .ce_swiperStart {
 	position:relative;
+	width: 100%;
```
no. I just added it because of some styling that was loaded inside my test.
I removed the width again.

Xirdion commented on src/Elements/WrapperStop.php:
```diff
@@ -86,16 +86,16 @@ protected function compile()
+            {
+                // set pagination to true, so it will be enabled
+                $arrParams['pagination'] = true;
+            }
```
Now there should be tabs.

Xirdion commented:
@fritzmg I updated the code to meet your needs.

fritzmg commented on src/Resources/public/contao-swiper.js:
```diff
@@ -0,0 +1,60 @@
+            new Swiper(sliderElement, params);
+        });
+    }
+})();
```
new line should be added for consistency :)

Xirdion replied:
Added the new line at the end of the file. :)

fritzmg commented on src/Resources/contao/dca/tl_content.php:
```diff
@@ -94,7 +94,7 @@
         'bullets' => 'Punkte',
         'fraction' => 'Bruchzahl',
-        'progress' => 'Fortschritt'
+        'progressbar' => 'Fortschritt'
```
Can you please change this back? This would create a BC break.

Xirdion replied:
I have undone the changes and added a query in `WrapperStop.php` so that *progress* turns into *progressbar* as this is the correct value in swiper.js.

fritzmg commented on src/Resources/contao/languages/de/tl_content.php:
```diff
@@ -13,7 +13,7 @@
 $GLOBALS['TL_LANG']['tl_content']['sliderSpaceBetween'] = ['Abstand zwischen den Slides','Abstand zwischen den Slides in Pixel.'];
 $GLOBALS['TL_LANG']['tl_content']['sliderEffect'] = ['Effekt','Effekt-Art beim Wechsel der Slides.'];
-$GLOBALS['TL_LANG']['tl_content']['sliderWrapperClass'] = ['Wrapper CSS Klasse','Die optionale CSS Klasse des Wrappers.'];
+$GLOBALS['TL_LANG']['tl_content']['sliderWrapperClass'] = ['Wrapper CSS Klasse','Die optionale CSS Klasse des Wrappers. Kann verwendet werden, wenn man z.B. durch eine Nachrichtenliste sliden möchte. Dann muss die Klassen des Listen-Containers angegeben werden (mod_newslist).'];
```
Can you add this in en/tl_content.php too please? ;)

Xirdion replied:
Fixed a typo in the german translation and added this to the english one.

fritzmg commented:
I still have one problem with the current concept. Originally, I initialized the slider in the stop wrapper element. This is the earliest opportunity the slider can be initialized, minimizing the chance of FOUC.

Now only the slider config is set - and the slider is initialized at the end of the body.

Xirdion commented:
I changed the code so that swiper.js and the custom-js to init a swiper-slider are loaded inside the head (`TL_JAVASCRIPT`). Inside the Template the function is called to build the slider.
I also added new classes to the slider-container, the pagination and the nav-buttons to handle multiple sliders on a page.

fritzmg commented:
👍 

I mean - we can still discuss this ;). What do you think? May be it is not necessary, but if you have a long document for example and a slider at the very top, the browser will probably show you the top of the document, before the bottom is reached during loading - with the slider in its uninitialized state.

On the downside, we have to load contao-swiper.js and swiper.js in the <head> - but I think this is reasonable for something like that.

Xirdion commented:
I think we should use it the way it is implemented now.
We still can change the behaviour if desired by someone. :)

fritzmg commented:
Alright :). I'll merge this and release it as `2.0.0`, since it's a major change.

Xirdion commented:
@fritzmg Should I update the README.md because jQuery is no dependency anymore?
At the moment the JavaScript is combined in the assets mentioned in #1. Is that okay?

fritzmg commented:
Ah yes please. Just add it to this PR. You can also add more documentation if you want (e.g. the stuff with the wrapper class is completely undocumented).

Xirdion commented:
I have updated the readme with some features of the slider. I hope it is okay :)

fritzmg commented on README.md:
```diff
@@ -7,6 +7,16 @@ Contao Swiper
+| *different slider-effects* | slide, fade, cube, coverflow, flip |
+| *different pagination-types* | bullets, fraction, progressbar |
+| *breakpoints* | change some settings on different viewports |
+| *custom wrapper-class* | Helpful when adding a *module* (e.g. newslist-module) inside the slider. You can define the `mod-newslist`-class as wrapper for the slides, so that it is possible to slide through the single news-entries. |
```
Should me `mod_newslist` not `mod-newslist` ;)

Xirdion replied:
corrected the class-name.

fritzmg commented on src/Elements/WrapperStop.php:
```diff
@@ -86,19 +86,19 @@ protected function compile()
+            		{
+                		// set pagination to true, so it will be enabled
+                		$arrParams['pagination'] = true;
+            		}
```
Ah, can you change the indents to tabs here please?

Xirdion replied:
changed the indents.

fritzmg commented on README.md:
```diff
@@ -7,6 +7,16 @@ Contao Swiper
 ## Usage
 
-__jQuery__ needs to be enabled in your page layout. This extension replaces the core slider with a Swiper implementation.
+__jQuery__ does **not** have to be enabled in your page layout.
```
I think you should leave the comment

> This extension replaces the core slider with a Swiper implementation.

since that is still true.

Xirdion replied:
It is still there: [in Line 10](https://github.com/fritzmg/contao-swiper/pull/3/files/4e5f0178641f7f667b0bc59d2e092100ca403829#diff-04c6e90faac2675aa89e2176d2eec7d8R10)

fritzmg replied:
Ha, sorry, did not see that 😁
